### PR TITLE
docs: write main documentation

### LINKED ISSUE #50: Re-write main documentation
## Description

The externalisation of the CLI app (https://github.com/benchttp/engine/pull/48) made obvious the lack of documentation on this repo: the Readme is **empty**.

## Tasks

Fill README.md with concise presentation of the repo and relevant information for dev (installation, ...)

### PR DESCRIPTION
**Pull request by GregoryAlbouy**
## Linked issues

- Resolves #50

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,12 +16,79 @@
 
 ## About
 
-:construction: WIP
+Benchttp engine is a Go library providing a way to perform benchmarks and tests
```
Comment (moreirathomas) [inline README.md:19]: Nitpick: it's not _really_ a lib in the way that it only has one way to be used. I think package don't bring that weight of composability with it.
```suggestion
Benchttp engine is a Go library package a way to perform benchmarks and tests
```

File: README.md [hunk 2]
```diff
@@ -16,12 +16,79 @@
...
+}`)
+
+    config, _ := configparse.JSON(jsonConfig)
+    report, _ := runner.New(nil).Run(context.Background(), config)
+
+    fmt.Println(report.Metrics.ResponseTimes.Mean)
+}
+```
+
+## Contributing: set up develpment environment
```
Comment (moreirathomas) [inline README.md:89]: Nitpick to mirror desktop readme.
```suggestion
## Development
```

Anyway, there's a typo 😛 
```suggestion
## Contributing: set up development environment
```

File: README.md (hunk 1 above)
Comment (c1-ra) [reply README.md:19]: I agree, package sound a little more suitable for me too.
```suggestion
Benchttp engine is a Go package providing a way to perform benchmarks and tests
```

Comment (GregoryAlbouy) [reply README.md:19]: To me it's still a library, as opposed to an executable application — I felt important to make that clear, as "package" is used for both world in Go.

Comment (moreirathomas) [reply README.md:19]: It makes sense too ! I leave judge on that :)

Comment (c1-ra) [reply README.md:19]: I have no strong argument about it either, so OK for library if you prefer it.

**Review by moreirathomas — APPROVED** [review]

**Review by c1-ra — APPROVED** [review]
Nice!

**Review by c1-ra — APPROVED** [review]
:+1:

**Pull request merged**